### ISSUE TITLE
Automatické propsání změn z údajů o svazku do popisu svazku bez využití znovuvygenerování

### ISSUE THREAD

**Issue by AlesBrozek** [NLCR/evidence.periodik#258]
Změnil jsem v "Údaje o svazku" mutaci z "Bez označení" na "Brno" a mutační vydání na dvě hvězdičky, objevila se hláška "Svazek byl úspěšně upraven", ale k úpravě nedošlo
![Snímek obrazovky 2024-10-10 v 23 08 47](https://github.com/user-attachments/assets/ee551747-a1fb-4b3e-8320-f275cfb0294d)

**Comment by AlesBrozek** [comment]
Teprve když jsem ťukl na "Upravit periodicitu" (abych dostal do popisu svazku chybějící podnázev a chybějící název přílohy) a dal vygenerovat svazek, dostal se do popisu svazku požadovaný údaj mutace Brno a mutační vydání 2 hvězdičky. Pokud se bude upravovat jen "Údaj o svazku" a periodicita bude v pořádku, měla by proběhnout změna v "Údaj o svazku" okamžitě.

**Comment by Kretiss** [comment]
@AlesBrozek Tyto úpravy v údajích o svazku se neprojevují ihned v popisu svazku, jak tomu bylo ve staré aplikaci (vím, že jsem to tam dodělával dodatečně a původně to tak nefungovalo). Projeví se to až po znovuvygenerování svazku přes periodicitu. Je to tedy žádoucí? Upraví to všechny záznamy, nehledě na to co v nich je

**Comment by AlesBrozek** [comment]
No pokud se upraví text v metodice, že se úpravy v údajích o svazku projeví až po vygenerování svazku přes periodicitu, tak je to možné. Časem zapomeneme, jak jsme to dělali ve staré aplikaci, a zvykneme si na novou :-)

**Comment by Kretiss** [comment]
@annavandasova @zabak

**Comment by AlesBrozek** [comment]
No, provedl jsem duplikaci a změny se propsaly do popisu svazku, datum vytvoření nového svazku je OK, ale datum upravení svazku je divné - viz printscreen. A po kliknutí na "ověřeno" se objeví divné datum vytvoření svazku - viz druhý printscreen

<img width="1440" height="900" alt="Image" src="https://github.com/user-attachments/assets/d574b4dc-089b-4afb-be02-93b4f6be71d1" />

<img width="1440" height="900" alt="Image" src="https://github.com/user-attachments/assets/efe3cb88-46af-47d6-b13c-cc592602273d" />

**Comment by JiriLoun-IQ** [comment]
Oba problemy by mely byt nyni opraveny, jednalo se o neuspesne resetovani nekterych servisnich atributu pri vytvoreni duplikatu

**Comment by AlesBrozek** [comment]
Teď jsem duplikoval z čk 1003250127 na čk 1003250468. U pravidelných příloh došlo k přiřazení čísla o jedno vyšší než mělo normální číslo: příloha z 4.10. má mít č. 40, ale dostane č. 235, takže výtisk z 5.10. dostane č. 236 místo správného 235 (při duplikování by se měl zkopírovat stav z detailního popisu svazku, z kterého se duplikuje, protože u něho jsou zohledněny změny ve vydávání např. na konci roku, kdy výtisk i s přílohou může vyjít v neděli)...

**Comment by AlesBrozek** [comment]
Nevím, co nazýváte "nová verze", ale ve verzi https://app.permonik.nkp.cz/ je funkce "duplikovat svazek", pomocí níž jsem od května 2025 zpracoval kolem 100 svazků, tedy 9000 výtisků. Tato funkce "duplikovat svazek" zkopíruje svazek včetně odlišných dat vydání výtisků (např. výtisků nevyšlých v sobotu, ale vyšlých výjimečně v neděli, či příloh Volno vyšlých výjimečně ve čtvrtek místo v pátek). Po spuštění funkce "duplikovat svazek" se v editaci vstupních dat vyplní mutace, mutační vydání, čárový kód a signatura. Vznikne nový svazek, který respektuje nepravidelnosti a výjimky uvedené ve svazku, z kterého se duplikovalo. V "Přehledu svazku" se objeví změněná signatura a nový čárový kód. Jediný problém je, že v "detailním popisu svazku" zůstává původní mutace (což při duplikování svazku, který měl mutaci Praha a nový svazek má také mutaci Praha, nevadí, ale při duplikaci svazku s mutací Praha na svazek s mutací Ostrava se musí ručně toto opravit u každého výtisku) a zůstává původní mutační vydání (nutno ručně opravit, ale existuje issue požadující dělat takové úpravy hromadně). Důkladnému diskutování se nebráním. Jak operátora donutit přepsat ČK není problém, protože v app.permonik.nkp.cz po spuštění funkce "duplikovat svazek" se v editaci vstupních dat pole mutační vydání a ČK vyprázdní (ještě by  bylo možné, aby se vyprázdnilo pole signatura, ale pro mne je výhodné, že v tom poli přepíši jen dvě čísla).

**Comment by JiriLoun-IQ** [comment]
Přidána duplikace cizích svazků

**Comment by JiriLoun-IQ** [comment]
Dobrý den, pokud vím, @Kretiss  dané změny ještě nenasadil na testovací prostředí. Až se tak stane, label Implemented přehodí na label To Test.

Mimochodem, tlačítko na duplikaci cizího svazku jsem dal do přehledu svazku, který se rozklikne z kalendáře/tabulky, úplně dolů.
